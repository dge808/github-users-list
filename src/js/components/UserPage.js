import React from 'react';
import moment from 'moment';
import { Link } from 'react-router';

const UserPage = ({ userInfo }) => {

    return (
        <main class="content-container">
            <h1 class="page-header">{userInfo.name}</h1>
            <div class="current-user-info">
                <img src={userInfo.avatar_url} class="user-avatar"/>

                <div class="about-user">
                    {userInfo.followers &&
                        <p>followers: {userInfo.followers}</p>}
                    {userInfo.following &&
                        <p>followers: {userInfo.following}</p>}
                    {userInfo.created_at &&
                        <p>created_at: {moment(userInfo.created_at).format('MMMM Do YYYY')}</p>}
                    {userInfo.company &&
                        <p>company: {userInfo.company}</p>}
                    {userInfo.email &&
                        <p>email: {userInfo.email}</p>}
                    {userInfo.location &&
                        <p>location: {userInfo.location}</p>}
                    {userInfo.blog &&
                        <p>blog: <a href={userInfo.blog}>{userInfo.blog}</a></p>}
                    {userInfo.bio &&
                        <p>bio: {userInfo.bio}</p>}
                </div>
            </div>
            <Link to="users/list" class="go-back-link">
                Go back to users list
            </Link>
            <div class="clearfix"></div>
        </main>
    )
};

export default UserPage;

UserPage.propTypes = {
    userInfo: React.PropTypes.object
};