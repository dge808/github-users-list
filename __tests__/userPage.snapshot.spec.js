import React from 'react';
import renderer from 'react-test-renderer';
import UserPage from '../src/js/components/UserPage';

const userInfo = {
    name: 'Alina',
    followers: 2,
    following: 2,
    created_at: "2015-07-31T05:24:19Z",
    company: null,
    email: "dge808@gmail.com",
    location: "Ukraine",
    blog: "",
    bio: null
};

describe('User Page(snapshot)', () => {
    it('should match User Page snapshot', () => {
        const component = renderer.create(<UserPage userInfo={userInfo}/>);
        const json = component.toJSON();
        expect(json).toMatchSnapshot();
    })
});