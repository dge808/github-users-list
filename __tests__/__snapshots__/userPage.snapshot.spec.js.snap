exports[`User Page(snapshot) should match User Page snapshot 1`] = `
<main
  className="content-container">
  <h1
    className="page-header">
    Alina
  </h1>
  <div
    className="current-user-info">
    <img
      className="user-avatar"
      src={undefined} />
    <div
      className="about-user">
      <p>
        followers: 
        2
      </p>
      <p>
        followers: 
        2
      </p>
      <p>
        created_at: 
        July 31st 2015
      </p>
      <p>
        email: 
        dge808@gmail.com
      </p>
      <p>
        location: 
        Ukraine
      </p>
      
    </div>
  </div>
  <a
    className="go-back-link"
    onClick={[Function]}
    style={Object {}}>
    Go back to users list
  </a>
  <div
    className="clearfix" />
</main>
`;
